src/api/app/services/staging/staging_project_creator.rb

```diff
@@ -0,0 +1,40 @@
+module Staging
+  class StagingProjectCreator
+    def initialize(staging_project_names, staging_workflow)
+      @staging_project_names = staging_project_names
+      @staging_workflow = staging_workflow
+    end
+
+    def perform
```

Comment:
why don't call it ```call``` which is "standard" in the service objects pattern in ruby? I don't think that ```perform``` is a bad name, but if you look all the others services that we have, I think all of them have similar API. 

One advantage to have it named to ```call``` is to be able to stub them with simple procs as in https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/controllers/webui/projects/meta_controller_spec.rb#L29 making easier to get rid of VCR cassettes..

Reply:
@vpereira, I will change the name of the method.

Reply:
> why don't call it `call` which is "standard" in the service objects pattern in ruby? I don't think that `perform` is a bad name, but if you look all the others services that we have, I think all of them have similar API.

I see a mixed picture there:
```
coolo@f102#api>git grep 'def call' app | wc -l
7
coolo@f102#api>git grep 'def perform' app | wc -l
28
```

Looks like all the calls are @vpereira 's :)

Reply:
https://hackernoon.com/service-objects-in-ruby-on-rails-and-you-79ca8a1c946e uses 'perform' while
https://medium.freecodecamp.org/service-objects-explained-simply-for-ruby-on-rails-5-a8cc42a5441f uses 'call' - I don't think there is something like a standard outside of OBS. And in OBS the standard I would say is 'perform'.

Reply:
> I see a mixed picture there:
> 
> ```
> coolo@f102#api>git grep 'def call' app | wc -l
> 7
> coolo@f102#api>git grep 'def perform' app | wc -l
> 28
> ```

@coolo looks like you are confusing jobs with services.. do you need any help?

```
vpereira@linux-y5ui:~/open-build-service/src/api/app/services> git grep "def call" | wc -l
6
vpereira@linux-y5ui:~/open-build-service/src/api/app/services> git grep "def perform" | wc -l
0
vpereira@linux-y5ui:~/open-build-service/src/api/app/services> 

```

Reply:
Ah, that explains it. Thanks

Reply:
> https://hackernoon.com/service-objects-in-ruby-on-rails-and-you-79ca8a1c946e uses 'perform' while
> https://medium.freecodecamp.org/service-objects-explained-simply-for-ruby-on-rails-5-a8cc42a5441f uses 'call' - I don't think there is something like a standard outside of OBS. And in OBS the standard I would say is 'perform'.

@coolo is really hard to argument here without folling in the "argument from authority" fallacy. But let's try:

https://www.toptal.com/ruby-on-rails/rails-service-objects-tutorial
https://medium.freecodecamp.org/service-objects-explained-simply-for-ruby-on-rails-5-a8cc42a5441f
http://chrisholtz.com/blog/organize-your-app-with-service-objects/

Reply:
I quote from there:

> The call method

> These services are invoked via the call() method. This method could be named anything - perform() > and process() are workable alternatives. However, I use call() because it is the same method name > used by ruby Proc and Lambda objects to invoke their content. This becomes more important when > applying functional programming principals to service objects. Also, call() can be invoked via 
> shortcut notation:

This gives enough reason.

Reply:
@vpereira, changed!